src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -1,22 +1,82 @@
 import logging
+from collections import defaultdict
 
 import sentry_sdk
 
-from sentry.utils import metrics
-from sentry.workflow_engine.models import Detector, Workflow
+from sentry import buffer
+from sentry.utils import json, metrics
+from sentry.workflow_engine.models import Detector, Workflow, WorkflowDataConditionGroup
+from sentry.workflow_engine.models.workflow import get_slow_conditions
 from sentry.workflow_engine.processors.action import evaluate_workflow_action_filters
+from sentry.workflow_engine.processors.data_condition_group import evaluate_condition_group
 from sentry.workflow_engine.processors.detector import get_detector_by_event
 from sentry.workflow_engine.types import WorkflowJob
 
 logger = logging.getLogger(__name__)
 
+WORKFLOW_ENGINE_BUFFER_LIST_KEY = "workflow_engine_delayed_processing_buffer"
+
+
+def get_data_condition_groups_to_fire(
+    workflows: set[Workflow], job: WorkflowJob
+) -> dict[int, list[int]]:
+    workflow_action_groups: dict[int, list[int]] = defaultdict(list)
+
+    workflow_ids = {workflow.id for workflow in workflows}
+
+    workflow_dcgs = WorkflowDataConditionGroup.objects.filter(
+        workflow_id__in=workflow_ids
+    ).select_related("condition_group", "workflow")
+
+    for workflow_dcg in workflow_dcgs:
+        action_condition = workflow_dcg.condition_group
+        evaluation, result = evaluate_condition_group(action_condition, job)
+
+        if evaluation:
+            workflow_action_groups[workflow_dcg.workflow_id].append(action_condition.id)
+
+    return workflow_action_groups
+
+
+def enqueue_workflows(
+    workflows: set[Workflow],
+    job: WorkflowJob,
+) -> None:
+    event = job["event"]
+    project_id = event.group.project.id
+    workflow_action_groups = get_data_condition_groups_to_fire(workflows, job)
+
+    for workflow in workflows:
+        buffer.backend.push_to_sorted_set(key=WORKFLOW_ENGINE_BUFFER_LIST_KEY, value=project_id)
+
+        if_dcgs = workflow_action_groups.get(workflow.id, [])
+        if not if_dcgs:
+            continue
+
+        if_dcg_fields = ":".join(map(str, if_dcgs))
+
+        value = json.dumps({"event_id": event.event_id, "occurrence_id": event.occurrence_id})
+        buffer.backend.push_to_hash(
+            model=Workflow,
+            filters={"project": project_id},
+            field=f"{workflow.id}:{event.group.id}:{if_dcg_fields}",
+            value=value,
+        )
+
 
 def evaluate_workflow_triggers(workflows: set[Workflow], job: WorkflowJob) -> set[Workflow]:
     triggered_workflows: set[Workflow] = set()
+    workflows_to_enqueue: set[Workflow] = set()
 
     for workflow in workflows:
         if workflow.evaluate_trigger_conditions(job):
             triggered_workflows.add(workflow)
+        else:
+            if get_slow_conditions(workflow):
+                # enqueue to be evaluated later
+                workflows_to_enqueue.add(workflow)
+
+    enqueue_workflows(workflows_to_enqueue, job)
```

Comment:
nit: save the couple of cpu cycles and only enqueue if we have something to enqueue
```suggestion
    if workflows_to_enqueue:
        enqueue_workflows(workflows_to_enqueue, job)
```